### PR TITLE
Assign page.guide_version via configuration scopes

### PR DESCRIPTION
**Pull request by dshevtsov**
## Purpose of this pull request

This pull request (PR) delegates generation of the `page.guide_version` parameter to native Jekyll means using configuration scopes, instead of expensive plugin with regex. This allows to reduce build time by ~1 min.

### CODE REVIEW

File: _plugins/page-params/page-baseurl-generator.rb [hunk 1]
```diff
@@ -6,29 +6,19 @@
 # This custom plugin dynamically sets and injects the page.baseurl variable
 # based on the page's destination.
 #
-# The hook introduces the page.baseurl and
-# page.guide_version data parameters for each page.
+# The hook introduces the page.baseurl parameter for each page.
 # For pages at guides/v2.x the page.baseurl parameter is set
 # as "{site.baseurl}/guides/v#{version}".
 # The {version} is taken from the 'guide_version' front matter parameter on the page;
```
Comment (jeff-matthews) [inline _plugins/page-params/page-baseurl-generator.rb:12]: Is it still possible to use `guide_version` in the metadata? If not, we should remove this comment.

Comment (dshevtsov) [reply _plugins/page-params/page-baseurl-generator.rb:12]: Yes, it's still possible. Front matter parameters always have privilege over default parameters assigned via scopes. (This is how we use layout at the moment.) For more information, refer to [Precedence in Jekyll](https://jekyllrb.com/docs/configuration/front-matter-defaults/#precedence).

**Comment by dshevtsov** [comment]
This change breaks `include/exclude versions` feature in TOC.
Converting to draft until resolved.

**Comment by dshevtsov** [comment]
running tests

**Pull request merged**